internal/shardpack/shardpack.go

```
@@ -264,6 +283,120 @@ func isPlanHashName(name string) bool {
+
+// GC removes result files that match the result naming patterns but no current
+// shard or plan. Nothing else in the directory is touched.
+func (w *writer) GC(root, scope, targetID string, plan contextprofile.ShardPlan) error {
+	resolvedRoot, err := w.deps.EvalSymlinks(root)
+	if err != nil {
+		return err
+	}
+	dir := ResultsDir(resolvedRoot, scope, targetID)
+	entries, err := w.deps.ReadDir(dir)
+	if err != nil {
+		return nil // nothing written yet
```

golangci-lint (2.12.2)

[error] error is not nil (line 374) but it returns nil

(nilerr)